modules/videoio/test/test_camera.cpp

```diff
@@ -104,5 +104,42 @@ TEST(DISABLED_videoio_camera, v4l_read_framesize)
 
     capture.release();
 }
+TEST(COUNT_OF_CAMERA_FRAMES, v4l_poll)
+{
+        VideoCapture cap1(0);
+        VideoCapture cap2(2);
+
+        std::vector<VideoCapture> VCM;
+
+        VCM.push_back(cap1);
+        VCM.push_back(cap2);
+
+        std::vector<int> State;
+
+        std::vector<int> CountOfStates(2, 0);
+
+        int iteration = 500;
+        for(int i = 0; i < iteration; ++i)
+        {
+            cv::VideoCapture::waitAny(VCM, State, 50);
+            for(unsigned int j = 0; j < State.size(); ++j)
+            {
+                Mat new_frame1;
+                if(State[j] == 1)
+                {
+                   //if(VCM[j].retrieve(new_frame1))
```

Comment:
Shouldn't it always be able to retrieve the frame if `State[j]` is 1? Then you can check it:
```.cpp
EXPECT_TRUE(VCM[j].retrieve(frame));
```